src/app/AuthenticationController.tsx

```diff
@@ -86,6 +120,16 @@ class AuthenticationController extends React.Component<AuthenticationControllerP
         <InitializingScreen errorMsg={AuthenticationController.PostLoginErrorMsg} />
       );
     } else if (this.state.stage === LoginStage.POST_LOGIN) {
+      // For OAuth/OpenID auth strategies, show/keep the initializing screen unless there
+      // is an error.
+      if (
+        !this.state.isPostLoginError &&
+        (authenticationConfig.strategy === AuthStrategy.openshift ||
+          authenticationConfig.strategy === AuthStrategy.openid)
+      ) {
```

Comment:
This multi-line conditional keeps getting repeated, perhaps for readability just add a simple private function isAuthStrategyOath() or something like that.

Reply:
I added the helper function. Not private, because this condition is on several files.